### ISSUE TITLE
TypeError: unsupported operand type(s) for +: 'NoneType' and 'int'

### ISSUE THREAD

**Issue by ianporada** [SapienzaNLP/maverick-coref#2]
Ran into a strange error when running on examples from OntoNotes, so sharing it here.

```python
words = [['I', 'will', 'have', 'to', 'see', 'the', 'nail', 'holes', 'in', 'his', 'hands', ',', 'put', 'my', 'finger', 'where', 'the', 'nails', 'were', ',', 'and', 'put', 'my', 'hand', 'into', 'his', 'side', '.'], ['Only', 'then', 'will', 'I', 'believe', 'it', '.', "''"], ['A', 'week', 'later', 'the', 'followers', 'were', 'in', 'the', 'house', 'again', ',', 'and', 'Thomas', 'was', 'with', 'them', '.']]

speaker = [['John', 'John', 'John', 'John', 'John', 'John', 'John', 'John', 'John', 'John', 'John', 'John', 'John', 'John', 'John', 'John', 'John', 'John', 'John', 'John', 'John', 'John', 'John', 'John', 'John', 'John', 'John', 'John'], ['John', 'John', 'John', 'John', 'John', 'John', 'John', 'John'], ['John', 'John', 'John', 'John', 'John', 'John', 'John', 'John', 'John', 'John', 'John', 'John', 'John', 'John', 'John', 'John', 'John']]

from maverick import Maverick

model = Maverick()
model.predict(words, speakers=speaker)
```

Gives the error below. Strangely works without speaker specified or when the speaker names are changed to other strings.

```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
[<ipython-input-11-ddf4a656ef6b>](https://localhost:8080/#) in <cell line: 1>()
----> 1 model.predict(words, speakers=speaker)

3 frames
[/usr/local/lib/python3.10/dist-packages/torch/utils/_contextlib.py](https://localhost:8080/#) in decorate_context(*args, **kwargs)
    113     def decorate_context(*args, **kwargs):
    114         with ctx_factory():
--> 115             return func(*args, **kwargs)
    116 
    117     return decorate_context

[/usr/local/lib/python3.10/dist-packages/maverick/models/maverick_model.py](https://localhost:8080/#) in predict(self, sample, singletons, add_gold_clusters, predefined_mentions, speakers)
    132         result["clusters_token_offsets"] = clusters_predicted
    133         result["clusters_char_offsets"] = None
--> 134         result["clusters_token_text"] = [
    135             [" ".join(tokens[span[0] : span[1] + 1]) for span in cluster] for cluster in clusters_predicted
    136         ]

[/usr/local/lib/python3.10/dist-packages/maverick/models/maverick_model.py](https://localhost:8080/#) in <listcomp>(.0)
    133         result["clusters_char_offsets"] = None
    134         result["clusters_token_text"] = [
--> 135             [" ".join(tokens[span[0] : span[1] + 1]) for span in cluster] for cluster in clusters_predicted
    136         ]
    137         result["clusters_char_text"] = None

[/usr/local/lib/python3.10/dist-packages/maverick/models/maverick_model.py](https://localhost:8080/#) in <listcomp>(.0)
    133         result["clusters_char_offsets"] = None
    134         result["clusters_token_text"] = [
--> 135             [" ".join(tokens[span[0] : span[1] + 1]) for span in cluster] for cluster in clusters_predicted
    136         ]
    137         result["clusters_char_text"] = None

TypeError: unsupported operand type(s) for +: 'NoneType' and 'int'
```

**Comment by ianporada** [comment]
I guess this has something to do with one of the words being `"''"`.

**Comment by ianporada** [comment]
Also not completely related but here's an interesting case where a predicted mention spans two sentences:

```python
words = [['He', 'confirmed', 'it', 'by', 'repeating', 'verses', 'from', 'the', 'noble', 'Koran', 'and', 'the', 'two', 'testimonies', '.'], ['He', 'will', 'be', 'envied', 'by', 'all', 'Arab', 'leaders', ',', 'both', 'living', 'and', 'dead', ',', 'for', 'this', 'honorable', 'martyrdom', ',', 'and', 'this', 'enormous', 'love', 'in', 'the', 'midst', 'of', 'hundreds', 'of', 'millions', 'of', 'Arabs', 'and', 'Muslims', 'all', 'over', 'the', 'different', 'parts', 'of', 'the', 'world', '.'], ['Saddam', 'Hussein', 'was', 'the', 'only', 'leader', 'in', 'the', 'modern', 'history', 'of', 'this', 'nation', 'who', 'went', 'to', 'the', 'guillotine', 'because', 'he', 'was', 'a', 'patriot', 'who', 'refused', 'occupation', 'and', 'surrender', 'to', 'the', 'invasion', ',', 'and', 'chose', 'resistance', '.']]
speakers = [['_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_'], ['_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_'], ['_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_', '_أسعد_']]

model.predict(words, speakers=speakers)

>> ... 'clusters_token_offsets': [((0, 0), (0, 15), (15, 15), (58, 59), (77, 77))] ...
```

**Comment by g185** [comment]
Hi @ianporada, thanks for using Maverick!

With the latest commit I should have fixed the two different bugs you noticed:
The first happened when a combination of non-closing brackets `"''"` and speaker name happened in the same sentence. the model mistakenly predicted the speaker, which is encoded in the sentence, as a mention. since this cannot happen, I fixed this problem just by adding a "None" check in the cluster offset reconstruction, reference: [util.py changes](https://github.com/SapienzaNLP/maverick-coref/commit/041b4466a5532e449e22e6672261206eb7ca1c22)
The second bug was due to EOS being wrongly set in the beginning of the following sentence.

Thanks a lot for noticing those two little bugs, I am available if you have any other problem or suggestions.
